pkg/eval/vals/equal_test.go

```diff
@@ -22,6 +23,14 @@ func TestEqual(t *testing.T) {
 		Args(true, false).Rets(false),
 
 		Args(1.0, 1.0).Rets(true),
+		Args(1.0, 1.1).Rets(false),
+		Args("1.0", 1.0).Rets(false),
+		Args(1, 1.0).Rets(false),
+		Args(1, 1).Rets(true),
+		Args(big.NewInt(1), big.NewInt(1)).Rets(true),
+		Args(big.NewInt(1), 1).Rets(false),
```

Comment:
Internally, Elvish never represents integers within the range of `int` as a `*big.Int`; this should test some actual big numbers; num_test.go has some utilities for doing this.